client/ayon_deadline/lib.py

```diff
@@ -43,3 +53,513 @@ def get_instance_job_envs(instance) -> "dict[str, str]":
+
+    OverrideTaskExtraInfoNames: Optional[bool] = field(
+        default=False)  # Default false
+
+    TaskExtraInfoName: Any = field(
+        default_factory=partial(DeadlineIndexedVar, "TaskExtraInfoName"))
+
+    OutputFilename: Any = field(
+        default_factory=partial(DeadlineIndexedVar, "OutputFilename"))
+    OutputFilenameTile: str = field(
+        default_factory=partial(DeadlineIndexedVar, "OutputFilename{}Tile"))
+    OutputDirectory: str = field(
+        default_factory=partial(DeadlineIndexedVar, "OutputDirectory"))
+
+    AssetDependency: str = field(
+        default_factory=partial(DeadlineIndexedVar, "AssetDependency"))
+
+    TileJob: bool = field(default=False)
+    TileJobFrame: int = field(default=0)
+    TileJobTilesInX: int = field(default=0)
+    TileJobTilesInY: int = field(default=0)
+    TileJobTileCount: int = field(default=0)
+
+    MaintenanceJob: bool = field(default=False)
+    MaintenanceJobStartFrame: int = field(default=0)
+    MaintenanceJobEndFrame: int = field(default=0)
+
+
+@dataclass
+class AYONDeadlineJobInfo(DeadlineJobInfo):
+    """Contains additional AYON variables from Settings for internal logic."""
+
+    # AYON custom fields used for Settings
+    UsePublished: Optional[bool] = field(default=None)
+    UseAssetDependencies: Optional[bool] = field(default=None)
+    UseWorkfileDependency: Optional[bool] = field(default=None)
+
+    def serialize(self):
+        """Return all data serialized as dictionary.
+
+        Returns:
+            OrderedDict: all serialized data.
+
+    """
+        def filter_data(a, v):
+            if isinstance(v, (DeadlineIndexedVar, DeadlineKeyValueVar)):
+                return False
+            if v is None:
+                return False
+            return True
+
+        serialized = asdict(self)
+        serialized = {
+            k: v for k, v in serialized.items()
+            if filter_data(k, v)
+        }
+
+        # Custom serialize these attributes
+        for attribute in [
+            self.EnvironmentKeyValue,
+            self.ExtraInfo,
+            self.ExtraInfoKeyValue,
+            self.TaskExtraInfoName,
+            self.OutputFilename,
+            self.OutputFilenameTile,
+            self.OutputDirectory,
+            self.AssetDependency
+        ]:
+            serialized.update(attribute.serialize())
+
+        for attribute_key in [
+            "LimitGroups",
+            "Whitelist",
+            "Blacklist",
+        ]:
+            serialized[attribute_key] = ",".join(serialized[attribute_key])
+
+        return serialized
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> 'AYONDeadlineJobInfo':
+
+        implemented_field_values = {
+            "ChunkSize": data["chunk_size"],
+            "Priority": data["priority"],
+            "MachineLimit": data["machine_limit"],
+            "ConcurrentTasks": data["concurrent_tasks"],
+            "Frames": data["frames"],
```

Comment:
This should fix the issue reported in https://github.com/ynput/ayon-deadline/pull/49/files#r1840677861
```suggestion
            "Frames": data.get("frames", ""),
```